metrics/open-telemetry-backend/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryBackend.scala

```diff
@@ -0,0 +1,79 @@
+package sttp.client3.opentelemetry
+
+import io.opentelemetry.api.common.{AttributeKey, Attributes}
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.api.trace.{Span, SpanKind, StatusCode, Tracer}
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.{TextMapPropagator, TextMapSetter}
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.monad.MonadError
+import sttp.monad.syntax._
+
+import scala.collection.mutable
+
+private class OpenTelemetryBackend[F[_], P](
+    delegate: SttpBackend[F, P],
+    tracer: Tracer
+) extends SttpBackend[F, P] {
+
+  private implicit val _monad: MonadError[F] = responseMonad
+  type PE = P with Effect[F]
+
+  def send[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
+    val carrier: mutable.Map[String, String] = mutable.Map().empty
+    val propagator: TextMapPropagator = W3CTraceContextPropagator.getInstance()
+    val setter: TextMapSetter[mutable.Map[String, String]] = (carrier, key, value) => carrier.update(key, value)
+    responseMonad
+      .eval {
+        val span: Span = tracer
+          .spanBuilder(s"HTTP ${request.method.method}")
+          .setSpanKind(SpanKind.CLIENT)
+          .startSpan
+        val ignored = span.makeCurrent()
+        propagator.inject(Context.current(), carrier, setter)
+        val attributes = Attributes.of(
+          AttributeKey.stringKey("http.method"),
+          request.method.method,
+          AttributeKey.stringKey("http.url"),
+          request.uri.toString(),
+          AttributeKey.stringKey("component"),
+          "sttp3-client"
+        )
+        Span.current.addEvent("Starting the work.", attributes)
```

Comment:
These event names look very artificial. We should come up with something better (what's the "standard" for http requests?) or make them configurable